torch_geometric/nn/conv/message_passing.py

```diff
@@ -224,6 +225,12 @@ def propagate(self, edge_index: Adj, size: Size = None, **kwargs):
                 and assumed to be quadratic.
                 This argument is ignored in case :obj:`edge_index` is a
                 :obj:`torch_sparse.SparseTensor`. (default: :obj:`None`)
+            decomposed_layer (integer, optional): The number of feature decomposition layers,
+                the default value is 1 (that is, the feature decomposition method is not used).
+                For the gather-scatter mode calculated on the CPU side, the feature decomposition
+                method can effectively reduce the peak memory usage and increase the calculation speed.
+                When the average node degree is higher, it is more effective to use this method
+                (for example, use Reddit dataset, achieves 2x - 3x speedup). More details in this paper: https://arxiv.org/abs/2104.03058 .
```

Comment:
Correct me if I am wrong here, but it looks like decomposition won't work on layers like [TransformerConv](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/conv/transformer_conv.html#TransformerConv) or [AGNNConv](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/conv/agnn_conv.html#AGNNConv), as the attention is computed based on `x_i` and `x_j` in `message` and decomposition sends a decomposed `x_i` and `x_j`. It'll be good to mention here or as a note the scenarios where decomposition won't work.

Reply:
Yes，you are right. My method is not effective for models ( Like: GAT, TransformerConv.....) based on attention mechanisms. My method mainly improves basic message passing models like GCN, GraphSage, GIN, etc.

Reply:
Yes，you are right. My method is not effective for models based on attention mechanism (such as GAT, TransformerConv ). My method mainly improves basic messaging models like GCN, GraphSage, GIN, etc.